v2/brigadier/src/jobs.ts

```diff
@@ -20,33 +61,177 @@ export class Job {
+   * primary container succeeded (exited with code 0), the Promise resolves; if the
+   * primary container failed (exited with nonzero code) or timed out, the Promise
+   * rejects. Sidecars do not affect success or failure of the job.
+   */
   public run(): Promise<void> {
     return Promise.resolve()
   }
 
+  /**
+   * Gets the job logs.
+   * 
+   * If the job has multiple containers, this aggregates the logs from them all.
+   */
   public logs(): Promise<string> {
     return Promise.resolve("skipped logs")
   }
+
+  /**
+   * Specifies a Runnable that consists of one or more containers. The
+   * image argument specifies the primary container, which determines
+   * how long the job runs and whether it is considered successful.
+   * By default, the primary container is simply
+   * executed (via its default entry point). Set the Job#primaryContainer#command
+   * property to run a specific command in the container instead.
+   * Other containers, specified via Job#sidecarContainers, are automatically
+   * terminated after the primary container completes.
+   * 
+   * (Note: This is equivalent to `new Job(...)`. It is provided so that
+   * script authors have the option of a consistent style for creating
+   * and composing jobs and groups.)
+   * 
+   * @param name The name of the job
+   * @param image The OCI image reference for the primary container
+   * @param event The event that triggered the job
+   */
+  public static container(
+    name: string,
+    image: string,
+    event: Event
+  ): Job {
+    return new Job(name, image, event)
+  }
+
+  /**
+   * Specifies a Runnable that consists of sub-Runnables (such as jobs
+   * or concurrent groups) running one after another.
+   * A new Runnable is started only when the previous one completes.
+   * The sequence completes when the last Runnable has completed (or when any
+   * Runnable fails).
+   * 
+   * @param runnables The work items to be run in sequence
+   */
+  public static sequence(...runnables: Runnable[]): SerialGroup {
+    return new SerialGroup(...runnables)
+  }
+
+  /**
+   * Specifies a Runnable that consists of sub-Runnables (such as jobs
+   * or sequential groups) running concurrently.
+   * When run, all Runnables are started simultaneously (subject to
+   * scheduling constraints).
+   * The concurrent group completes when all Runnables have completed.
+   * 
+   * @param runnables The work items to be run in parallel
+   */
+  public static concurrent(...runnables: Runnable[]): ConcurrentGroup {
+    return new ConcurrentGroup(...runnables)
+  }
 }
 
 export class Container {
+  /** The OCI image that the container should run. */
   public image: string
+  /**
+   * Specifies under what conditions the host node should re-fetch an image
+   * that is already in its local cache. The permitted values are as IfNotPresent
+   * (the default) and Always.
+   */
   public imagePullPolicy = "IfNotPresent"
```

Comment:
Non-blocking for this PR-- certainly not an issue introduced by this PR-- should we try to use an enum or constant here?

Reply:
We could use a TypeScript union type of strings, which would give the JS folks a familiar stringy interface (and compatibility with v1?) but provide intellisense and compilet-time validation to the TS folks:

```
type ImagePullPolicy = "Always" | "IfNotPresent";
public imagePullPolicy: ImagePullPolicy = "IfNotPresent";
```

Agree out of scope for this PR though.

Reply:
That looks good... I'm learning a lot of TS/JS from your suggestions.